update doc for max-user-connections (#20488)
pingcap/docs: system-variables.md

- In the `SESSION` scope, this variable is read-only.
- This variable is compatible with MySQL.

> ### `max_user_connections` <span class="version-mark">New in v9.0.0</span>

Reviewer: ```suggestion
### max_user_connections <span class="version-mark">New in v9.0.0</span>
```

- Scope: GLOBAL
- Persists to cluster: Yes
- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
- Type: Integer
- Default value: `0`
- Range: `[0, 100000]`

> - This variable indicates the maximum connection number for a user allowed to connect to a tidb-server instance in TiDB and is used for resource control.

Reviewer: ```suggestion
- This variable controls the maximum number of connections a user can establish to a TiDB server instance. It is used for resource control.
```

> - The default value `0` means there is no limit for user connection. When the value is greater than `0` and the number of user connections reaches this value, the TiDB server will reject the user's new connection.

Reviewer: ```suggestion
- The default value `0` means there is no limit for user connections. When the value is greater than `0` and the number of user connections reaches this value, the TiDB server will reject the user's new connection.
```

> - If the value of this variable exceeds [`max_connections`](/tidb-configuration-file.md#max_connections), TiDB uses `max_connections` to limit the maximum number of connections a single user can establish. For example, if `max_user_connections` of a user is set to `2000`, but `max_connections` is `1000`, the user can actually establish up to `1000` connections to a TiDB server instance.

Reviewer 2: ```suggestion
- If the value of this variable exceeds [`max_connections`](https://docs.pingcap.com/tidb/stable/tidb-configuration-file#max_connections), TiDB uses `max_connections` to limit the maximum number of connections a single user can establish. For example, if `max_user_connections` of a user is set to `2000`, but `max_connections` is `1000`, the user can actually establish up to `1000` connections to a TiDB server instance.
```

- Default value: `""`
- This variable is used to control some internal behaviors of the optimizer.
- The optimizer's behavior might vary depending on user scenarios or SQL statements. This variable provides a more fine-grained control over the optimizer and helps to prevent performance regression after upgrading caused by behavior changes in the optimizer.

> - For a more detailed introduction, see [Optimizer Fix Controls](https://docs-archive.pingcap.com/tidb/v7.2/optimizer-fix-controls).

Reviewer 2: ```suggestion
- For a more detailed introduction, see [Optimizer Fix Controls](https://docs.pingcap.com/tidb/stable/optimizer-fix-controls).
```
